Changelog for 3.7.3 (#4592)
apollographql/apollo-kotlin: CHANGELOG.md

Change Log
==========

# Version 3.7.3

_2022-12-20_

This release contains a handful of bug fixes and improvements, and also discontinues the legacy JS artifacts. 

Many thanks to @StefanChmielewski and @chao2zhang for contributing to the project! 🧡

> ## ⚙️ JS legacy artifacts (#4591)

Reviewer: Sorry missed that but we could be more explicit there for those like me that just scan the `h2` titles

```suggestion
## ⚙️ Removed JS legacy artifacts (#4591)
```

> Historically, Kotlin Multiplatform has had 2 formats of JS artifacts: [Legacy and IR](https://kotlinlang.org/docs/js-ir-compiler.html), and Apollo Kotlin has been publishing both. However, the Legacy format is about to [be deprecated with Kotlin 1.8](https://kotlinlang.org/docs/whatsnew-eap.html#kotlin-js) and moreover using the Legacy artifact causes some [hanging](https://github.com/apollographql/apollo-kotlin/issues/4590) in the browser. That is why starting with this release, only the IR artifacts will be published. Please reach out if this causes any issue in your project.

Reviewer: Made the wording less specific since we don't know how widespread this is: 

```suggestion
Historically, Kotlin Multiplatform has had 2 formats of JS artifacts: [Legacy and IR](https://kotlinlang.org/docs/js-ir-compiler.html), and Apollo Kotlin has been publishing both. However, the Legacy format is about to [be deprecated with Kotlin 1.8](https://kotlinlang.org/docs/whatsnew-eap.html#kotlin-js) and moreover we've seen [issues](https://github.com/apollographql/apollo-kotlin/issues/4590) when using the Legacy artifact in the browser. That is why starting with this release, only the IR artifacts will be published. Please reach out if this causes any issue in your project.
```
